### ISSUE TITLE
[UPRISE]When training the uprise,a problem happened.

### ISSUE THREAD

**Issue by zhouchang123** [microsoft/LMOps#238]
Here is the config and the log info.
![image](https://github.com/microsoft/LMOps/assets/61148892/ac8c172b-30f8-43f8-865b-59bf315fa7f1)
Here is the wrong message.
![image](https://github.com/microsoft/LMOps/assets/61148892/40bbbeca-7ceb-44e5-b8ae-171ab67a898a)
Everythings seems correct but until the first epoch goes end.
What's the reason of this problem?

**Comment by cdxeve** [comment]
The data amount is too small for validation, try increasing the data amount.

**Comment by zhouchang123** [comment]
Do you mean increase the batch-size?

**Comment by cdxeve** [comment]
<img width="821" alt="image" src="https://github.com/microsoft/LMOps/assets/26034310/8e3c45b2-2639-4a13-8c35-067849023c61">
I mean the number of validation samples,  from the above figure, your total data sample num is only 249, try adding more samples in the scored files.

**Comment by zhouchang123** [comment]
batch_size: 1
model_name: "EleutherAI/gpt-neo-2.7B"
output_train_file: ???
output_valid_file: ???
example_file: ??? # files containing the task inputs and its corresponding sampled prompt ids
task_name: ???
prompt_pool_path:
cache_dir: ???
max_length: 2048 # max seq length
generate_max_len: 100 # max length to be generated

dataset_reader:
  _target_: src.dataset_readers.scorer_dsr.ScorerDatasetReader
  example_file: ${example_file}
  model_name: ${model_name}
  task_name: ${task_name} 
  prompt_pool_path: ${prompt_pool_path}
  cache_dir: ${cache_dir}
  max_length: ${max_length}

Above is the config in score.yaml,it seems that no config about size.
Whether  I should to modify random_finder.yaml  like this?
![image](https://github.com/microsoft/LMOps/assets/61148892/bd9a732a-1431-444b-83c2-e6fcef2959ac)

**Comment by cdxeve** [comment]
The number of data samples of the task `rte` is too small, one way is to change to another task or include more tasks data in the the json files; another easy way is just to repeat the data samples in the validation json file.

**Comment by zhouchang123** [comment]
![image](https://github.com/microsoft/LMOps/assets/61148892/739a4485-c131-4ccb-b7d2-07fac3eb5696)
If just repeat the data,it still goes wrong.

**Comment by cdxeve** [comment]
<img width="771" alt="image" src="https://github.com/microsoft/LMOps/assets/26034310/8c8de57f-ff3b-41e9-9ccd-ffb9b6278506">

**Comment by zhouchang123** [comment]
![image](https://github.com/microsoft/LMOps/assets/61148892/210602af-e83e-4a73-80a7-10ad388444eb)
The problem still happened.

**Comment by cdxeve** [comment]
Can you show me the full log? It seems that the file is empty?

**Comment by zhouchang123** [comment]
[train.log](https://github.com/user-attachments/files/16103841/train.log)

**Comment by cdxeve** [comment]
I wonder if there is an error with your parallel settings again. Did you add CUDA_VISIBLE_DEVICES='xxx' before your running command? It seems there are 3 GPUs, but the world size is set to 1. Possibly, some GPUs are not receiving any data input for validation, so the vector count is 0. I suggest using only one GPU by setting CUDA_VISIBLE_DEVICES='0' throughout the process to avoid such errors.

<img width="798" alt="image" src="https://github.com/microsoft/LMOps/assets/26034310/dc833867-d81e-4c7d-a345-08ca3a6710b9">

**Comment by zhouchang123** [comment]
After  set CUDA_VISIBLE_DEVICES='0' ,a new error occured.
[newtrain.log](https://github.com/user-attachments/files/16104368/newtrain.log)

**Comment by zhouchang123** [comment]
After adding the code under,I found that all the tensor is on cpu and the model is on cuda:0.
What's the reason of this phenomenon?
![image](https://github.com/microsoft/LMOps/assets/61148892/2e145e6a-e8fe-4d91-afa0-ad53219adbc3)
@cdxeve

**Comment by cdxeve** [comment]
I did not encounter this error when running the code, and I'm not sure whether it is caused by a mismatch of devices or packages. You may try manually loading the variables to the same device as the models.

**Comment by zhouchang123** [comment]
Could you please tell me about where `trainer.run_train() `and `trainer.validate_average_rank()` move variables to cuda ?

**Comment by cdxeve** [comment]
Hi, the training code is largely borrowed from the official [facebookresearch/DPR](https://github.com/facebookresearch/DPR) repo. I did not modify the core trainer part, so you may need to debug and find the error yourself step-by-step.

**Comment by zhouchang123** [comment]
Maybe the code is partly loss in `validate_average_rank()` .The variables didn't move to `cfg.device` .After add these code,the program run successfully.
                if torch.is_tensor(q_ids):
                    q_ids=q_ids.to(cfg.device)
                if torch.is_tensor(q_segments):
                    q_segments=q_segments.to(cfg.device)
                if torch.is_tensor(q_attn_mask):
                    q_attn_mask=q_attn_mask.to(cfg.device)
                if torch.is_tensor(ctx_ids_batch):
                    ctx_ids_batch=ctx_ids_batch.to(cfg.device)
                if torch.is_tensor(ctx_seg_batch):
                    ctx_seg_batch=ctx_seg_batch.to(cfg.device)
                if torch.is_tensor(ctx_attn_mask):
                    ctx_attn_mask=ctx_attn_mask.to(cfg.device)
